### PR TITLE
Changed hierarchy Taggable<-Shred to Taggable<-Cluster<-Shred.

### PR DESCRIPTION
**Pull request by xa4a**
There's a need for a better name for Taggable. In the meantime, as the title says, the hierarchy changes as discussed previously to allow multi-shred stitching/tagging.

[Review on Reviewable](https://reviewable.io/reviews/dchaplinsky/unshred-tag/70)

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -69,26 +69,26 @@ def index():
...
+    if not item:
         abort(404)
 
     if request.method == "POST":
         # TODO: helper
         tags = set(map(unicode.lower, request.form.getlist("tags")))
 
-        shred_tag = shred.get_user_tags(g.user)
-        if not shred_tag:
+        user_tags = shred.get_user_tags(g.user)
```
Comment (dchaplinsky) [inline app.py]: item?

Comment (dchaplinsky) [inline app.py]: ping!

File: models/shreds.py [hunk 2]
```diff
@@ -52,9 +52,38 @@ def get_auto_tags(self):
...
+class Cluster(EmbeddedDocument):
+    """Cluster of one or more shreds.
+
+    Clusters are immutable once created.
+
+    Shred membership described with a ClusterMember embedded document, which
+    contains a reference to the shred and its relative position and angle in a
+    cluster."""
+    id = StringField(required=True, max_length=36, primary_key=True)
+    members = ListField(EmbeddedDocumentField(ClusterMember))
```
Comment (dchaplinsky) [inline models/shreds.py]: Cannot cluster become a cluster member of another cluster?

File: models/shreds.py [hunk 3]
```diff
@@ -52,9 +52,38 @@ def get_auto_tags(self):
...
+    @property
+    def features(self):
+        # TODO: persist features on creation.
+        return self.members[0].shred.features
+
+
 class Taggable(Document):
     id = StringField(max_length=200, default='', primary_key=True)
-    object = EmbeddedDocumentField(Shred)
+    object = EmbeddedDocumentField(Cluster)
```
Comment (dchaplinsky) [inline models/shreds.py]: Do we really need to separate taggable from cluster now?

File: models/shreds.py (hunk 2 above)
Comment (xa4a) [inline models/shreds.py]: Not according to the current design.

File: models/shreds.py (hunk 3 above)
Comment (xa4a) [inline models/shreds.py]: Merging two sounds reasonable and I'd like to do it in the next step.

File: app.py (hunk 1 above)
Comment (xa4a) [inline app.py]: Done.

**Comment by dchaplinsky** [comment]
Good to go then!

**Pull request merged**